it/api-v1/city_service_test.go

```diff
@@ -0,0 +1,229 @@
+////go:build integration
+
+package api_v1
+
+import (
+	"net/http"
+	"testing"
+	"time"
+
+	. "github.com/onsi/gomega"
+
+	"github.com/NikolNikolaeva/project_weather/it/testbed"
+	"github.com/NikolNikolaeva/project_weather/it/testbed/generated/client"
+)
+
+// GET: /api/cities -> OK (No cities)
+func Test_givenThatNoCitiesExist_whenGetCitiesIsInvoked_thenResponseIsOK(t *testing.T) {
+	testbed.Prepare(t, 30*time.Second, func(t *testing.T, tb *testbed.Runner) {
+		defer tb.Reset()
+
+		// execute
+		result, response, err := tb.APIClient.CityAPI.
+			GetAll(tb.Context).Execute()
+
+		// assert
+		Expect(err).ShouldNot(HaveOccurred())
+		Expect(result).To(BeNil())
+		Expect(response).To(HaveHTTPStatus(http.StatusOK))
+		Expect(result).To(BeNil())
+	})
+}
+
+// GET: /api/cities -> OK  -> 1 city
+func Test_givenThatNoCitiesExist_whenGetAllIsInvoked_thenResponseIsOK(t *testing.T) {
+	testbed.Prepare(t, 30*time.Second, func(t *testing.T, tb *testbed.Runner) {
+		defer tb.Reset()
+
+		// prepare
+		city := tb.CreateCity(testbed.NewTestCity())
+
+		// execute
+		result, response, err := tb.APIClient.CityAPI.
+			GetAll(tb.Context).Execute()
+
+		// assert
+		Expect(err).ShouldNot(HaveOccurred())
+		Expect(response.StatusCode).To(Equal(http.StatusOK))
+		Expect(result).ToNot(BeNil())
+		Expect(result).To(Equal([]client.City{testbed.AsCityDto(city)}))
+	})
+}
+
+// DELETE: /api/cities/{id} -> OK
+func Test_givenCityExists_whenDeleteByIdIsInvoked_thenResponseIsOK(t *testing.T) {
+	testbed.Prepare(t, 30*time.Second, func(t *testing.T, tb *testbed.Runner) {
+		defer tb.Reset()
+
+		// prepare
+		city := tb.CreateCity(testbed.NewTestCity())
+
+		// execute
+		response, err := tb.APIClient.CityAPI.DeleteById(tb.Context, city.ID).Execute()
+
+		// assert
+		Expect(err).ShouldNot(HaveOccurred())
+		Expect(response).To(HaveHTTPStatus(http.StatusOK))
+
```

Comment:
It's a good idea to also check that the DB is empty afterwards (city indeed got deleted)